tests/unit/amo/components/TestUserProfile.js

```diff
@@ -0,0 +1,321 @@
+        username: 'tofumatt',
+      },
+    });
+    const root = renderUserProfile({ store });
+
+    expect(root.find('.UserProfile-user-since')).toHaveLength(1);
+    expect(root.find('.UserProfile-user-since').children())
+      .toHaveText('Aug 15, 2000');
+  });
+
+  it('renders LoadingText for account creation date while loading', () => {
+    const root = renderUserProfile({ params: { username: 'not-tofu' } });
+
+    expect(root.find('.UserProfile-user-since')).toHaveLength(1);
+    expect(root.find('.UserProfile-user-since').find(LoadingText))
+      .toHaveLength(1);
+  });
+
+  it("renders the user's number of add-ons", () => {
+    const { store } = dispatchSignInActions({
+      userProps: {
+        num_addons_listed: 70,
+        username: 'tofumatt',
+      },
+    });
+    const root = renderUserProfile({ store });
+
+    expect(root.find('.UserProfile-number-of-addons')).toHaveLength(1);
+    expect(root.find('.UserProfile-number-of-addons').children())
+      .toHaveText('70');
+  });
+
+  it('renders LoadingText for number of add-ons while loading', () => {
+    const root = renderUserProfile({ params: { username: 'not-tofu' } });
+
+    expect(root.find('.UserProfile-number-of-addons')).toHaveLength(1);
+    expect(root.find('.UserProfile-number-of-addons').find(LoadingText))
+      .toHaveLength(1);
+  });
+
+  it("renders the user's average add-on rating", () => {
+    const { store } = dispatchSignInActions({
+      userProps: {
+        average_addon_rating: 4.1,
+        username: 'tofumatt',
+      },
+    });
+    const root = renderUserProfile({ store });
+
+    expect(root.find('.UserProfile-rating-average')).toHaveLength(1);
+    expect(root.find('.UserProfile-rating-average').find(Rating))
+      .toHaveProp('rating', 4.1);
+  });
+
+  it('renders LoadingText for average add-on rating while loading', () => {
+    const root = renderUserProfile({ params: { username: 'not-tofu' } });
+
+    expect(root.find('.UserProfile-rating-average')).toHaveLength(1);
+    expect(root.find('.UserProfile-rating-average').find(LoadingText))
+      .toHaveLength(1);
+  });
+
+  it("renders the user's biography", () => {
+    const { store } = dispatchSignInActions({
+      userProps: {
+        biography: 'Not even vegan!',
+        username: 'tofumatt',
+      },
+    });
+    const root = renderUserProfile({ store });
+
+    expect(root.find('.UserProfile-biography').render().html())
+      .toContain('Not even vegan!');
+  });
+
+  it('omits a null biography', () => {
+    const { store } = dispatchSignInActions({
+      userProps: {
+        biography: null,
+        username: 'tofumatt',
+      },
+    });
+    const root = renderUserProfile({ store });
+
+    expect(root.find('.UserProfile-biography')).toHaveLength(0);
+  });
+
+  it('omits an empty biography', () => {
+    const { store } = dispatchSignInActions({
+      userProps: {
+        biography: '',
+        username: 'tofumatt',
```

Comment:
The `username` is repeated everywhere to sync up with `params`. You could make a simple change to DRY it up:

```js
const { store } = dispatchSignInActions({
  userProps: defaultUserProps({
    biography: '',
  }),
});
```
Or you could wrap `dispatchSignInActions()` in a helper.

Reply:
Nice, will do!